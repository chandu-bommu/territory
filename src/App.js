import React, { Component } from 'react';
import logo from './assets/India_territory_map.png';
import './App.css';
import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
import { Dropdown, DropdownMenuItemType } from 'office-ui-fabric-react/lib/Dropdown';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Territories</h1>
        </header>
        <Dropdown
          placeHolder='Select an Option'
          label='Basic uncontrolled example:'
          id='Basicdrop1'
          ariaLabel='Basic dropdown example'
          options={
            [
              { key: 'Header', text: 'Actions', itemType: DropdownMenuItemType.Header },
              { key: 'A', text: 'Option a' },
              { key: 'B', text: 'Option b' },
              { key: 'C', text: 'Option c', disabled: true },
              { key: 'D', text: 'Option d' },
              { key: 'E', text: 'Option e' },
              { key: 'divider_2', text: '-', itemType: DropdownMenuItemType.Divider },
              { key: 'Header2', text: 'People', itemType: DropdownMenuItemType.Header },
              { key: 'F', text: 'Option f' },
              { key: 'G', text: 'Option g' },
              { key: 'H', text: 'Option h' },
              { key: 'I', text: 'Option i' },
              { key: 'J', text: 'Option j' },
            ]
          }
          componentRef={ this._basicDropdown }
        />
        <PrimaryButton
          text='Set focus'
          onClick={ this._onSetFocusButtonClicked }
        />        
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
