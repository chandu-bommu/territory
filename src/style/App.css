.App {
    text-align: center;
}

.App-logo {
    height: 100px;
}

.App-header {
    background-color: #0078d4;
    height: 80px;
    padding: 20px;
    color: white;
    background-image: url("../assets/header-background.jpg");
    background-size: 470px 370px;
}

.App-title {
    font-size: 1.0em;
    vertical-align: middle;
}

.appComponent {
    font-size: large;
    align-content: center;
    max-width: 50%;
}

.countryTable {
    padding-top: 30px;
}

.countryForm {
    padding-top: 3%;
}

.userDetails {
    padding-top: 50px;
}

table {
    font-family: arial, sans-serif;
    border-collapse: collapse;
    width: 200%;
}

#addButton,
#clearButton {
    margin: 20px 2px 2px 2px !important;
}

.countryTable th {
    border: 1px solid #dddddd;
    text-align: center;
    padding: 8px;
    background-color: #0078d4;
    color: #FFF;
}

.countryTable tr:nth-child(even) {
    background-color: #dddddd;
}

.userImage {
    background: #cbe5ff;
    box-shadow: 2px 2px 4px 4px lightgrey;
    width: 360px;
    height: 200px;
}

.userInfo {
    background: #FFF;
    box-shadow: 2px 2px 4px 4px lightgrey;
    width: 360px;
    height: 50px;
}

.userCard {
    padding-top: 5px;
}

#idSpan {
    padding-right: 70px;
}

.header {
    width: 200%;
    padding-bottom: 10px;
    font-size: 24px;
}

.contentDiv {
    width: 50%;
    height: auto;
    margin: 0 auto;
    padding: 10px;
    position: relative;
}