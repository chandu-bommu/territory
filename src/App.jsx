import React, { Component } from 'react';
import logo from './assets/India_territory_map.png';
import './style/App.css';
import Territory from './components/Territory.jsx';
import UserDetails from './components/UserDetails.jsx';

class App extends Component {
  render = () => {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Territories</h1>
        </header>
        <Territory />
        <UserDetails />
      </div>
    );
  }
}

export default App;
