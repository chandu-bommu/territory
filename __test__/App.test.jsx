import React from 'react';
import App from '../src/App.jsx';
import { shallow, mount } from 'enzyme';

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
    // On the first run of this test, Jest will generate a snapshot file automatically.
  });
});
