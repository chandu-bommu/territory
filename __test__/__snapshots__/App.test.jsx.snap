// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App"
  >
    <header
      className="App-header"
    />
    <div
      className="contentDiv"
    >
      <Territory />
      <UserDetails />
    </div>
  </div>,
  "nodes": Array [
    <div
      className="App"
    >
      <header
        className="App-header"
      />
      <div
        className="contentDiv"
      >
        <Territory />
        <UserDetails />
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <App />,
    "_forcedUpdate": false,
    "_instance": App {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="App"
    >
      <header
        className="App-header"
      />
      <div
        className="contentDiv"
      >
        <Territory />
        <UserDetails />
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
